### PR TITLE
Translation layer aggregation operation refactor

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Some cleanup and remove relationshipAggregationField in favour of OperationField

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3731.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/translate/queryAST/ast/fields/OperationField.ts [hunk 1]
```diff
@@ -17,31 +17,30 @@
...
     }
 
-    public get children(): QueryASTNode[] {
-        return [this.operation];
-    }
-
     public getProjectionField(): Record<string, Cypher.Expr> {
-        return { [this.alias]: this.projectionVariable };
+        if (!this.projectionExpr) {
+            throw new Error("Projection expression of operation not availabe (has transpiled been called)?");
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/fields/OperationField.ts:36]:
```suggestion
            throw new Error("Projection expression of operation not available (has transpiled been called)?");
```

File: packages/graphql/src/translate/queryAST/ast/operations/AggregationOperation.ts [hunk 2]
```diff
@@ -160,17 +155,23 @@ export class AggregationOperation extends Operation {
 
     // TODO: remove
     public transpile2({ returnVariable, parentNode }: OperationTranspileOptions): Cypher.Clause[] {
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/operations/AggregationOperation.ts:157]: can `transpile2` be removed?

**Review by MacondoExpress — APPROVED** [review]
Nice!

**Pull request merged**